torch/csrc/jit/tracer.cpp

```diff
@@ -1,18 +1,32 @@
-#include "Python.h"
 #include "torch/csrc/jit/tracer.h"
 
 #include "torch/csrc/autograd/variable.h"
 #include "torch/csrc/autograd/function.h"
 #include "torch/csrc/autograd/python_engine.h"
 #include "torch/csrc/autograd/functions/special.h"
-#include "torch/csrc/utils/auto_gil.h"
-#include "torch/csrc/utils/python_strings.h"
-
-#include <frameobject.h>
-#include <patchlevel.h>
 
 namespace torch { namespace jit { namespace tracer {
 
+// Python interpreter retrieval routine adapted from
+// https://stackoverflow.com/a/8706144
+std::string getPythonInterpreterStackTrace() {
```

Comment:
nit: it seems you're never using the string, but always wrap it in `SourceLocation`. Maybe just return that object from this function